Homework/homework2/task1_text.py

```diff
@@ -0,0 +1,133 @@
+import io
+import string
+from typing import List, Mapping, Tuple
+
+
+def u_decoder(line: str) -> str:
+    """Decode ascii chars."""
+    return line.encode().decode("unicode-escape")
+
+
+def decoded_line_without_punc(line: str) -> str:
+    """
+    Decode, remove punctuation from the string.
+
+    Add a long dash and angle quotes to a punctuation.
+    """
+    punctuation = string.punctuation + "\u2014" + "\u00bb" + "\u00ab"
+    map_char = str.maketrans(punctuation, " " * len(punctuation))
+    return u_decoder(line).translate(map_char).casefold()
+
+
+def _find_word_max_unique(word: str, max_word: str, max_len: int) -> Tuple[str, int]:
+    cur_len = len(set(word))
+    if max_len < cur_len:
+        max_len = cur_len
+        max_word = word
+    return max_word, max_len
+
+
+def _get_longest_diverse_words(file: io.StringIO()) -> List[str]:
+    long_words = []
+    for line in file:
+        max_word = ""
+        max_len = 0
+        words = decoded_line_without_punc(line).strip().split()
+        for word in words:
+            max_word, max_len = _find_word_max_unique(word, max_word, max_len)
+        long_words.append(max_word)
+    long_words.sort(key=lambda s: len(s), reverse=True)
+    file.close()
```

Comment:
Okay, but I can't understand where to close the file in this case? I need to open it, then put it as argument into the function with logic, then close the file and return result of the logic function. But flake8 argues that "R504 you shouldn`t assign value to variable if it will be use only as return value"..

Reply:
`.close()` on a file it is a bad practice.  In 99% of cases you should use `with`.

https://github.com/ValeriL/Epam_Python/pull/6#discussion_r521674798